lib/cdo/rack/attack.rb

```diff
@@ -1,60 +1,118 @@
+require_relative '../../dynamic_config/dcdo'
 require 'rack/attack'
 require 'rack/attack/path_normalizer'
 require 'active_support/core_ext/numeric/time'
 
-class Rack::Attack
-  # Rack::Attack returns a 429 for requests that exceed the rate limit.
+# A configuration object for setting and dynamically updating the RackAttack limits based
+# on DCDO/CDO configuration.
+class RackAttackConfigBase
 
-  redis_url = CDO.geocoder_redis_url || 'redis://localhost:6379'
-  cache.store = StoreProxy::RedisStoreProxy.new(Redis.new(url: redis_url))
+  # Configure rack attack to the use the specified cache store and dynamically update
+  # its configuration.
+  def initialize(cache_store)
+    Rack::Attack.cache.store = cache_store
+    DCDO.add_change_listener(self)
+    update_limits
+  end
 
-  # Don't strip trailing slashes.
-  remove_const(:PathNormalizer)
-  PathNormalizer = FallbackPathNormalizer
+  # A factory method that creates a RackAttackConfig backed by Redis.
+  def self.create
```

Comment:
Since factories don't usually have side effects, it's worth noting in comments that creating a RackAttackConfig also updates the limits (and cache) used by Rack::Attack.